src/useErrorBoundary.ts

```diff
@@ -2,15 +2,45 @@ import { useContext, useMemo, useState } from "react";
 import { assertErrorBoundaryContext } from "./assertErrorBoundaryContext";
 import { ErrorBoundaryContext } from "./ErrorBoundaryContext";
 
+type StateError<TError> =
+  | (TError & { _suppressLogging: true })
+  | (Error & { _suppressLogging: true });
```

Comment:
I am sorry, but I don't quite get this one. Why removing the generic parameter from `suppressLoggingForError` would be of help? I noticed that in the generic parameter of `useErrorBoundary`, `<TError = any>`, so I assume `TError` can be very flexible yet maintaining it can assure some basic type check. Please shed more light on this.